fileparsers/src/RegisterInfoMap.cpp

```diff
@@ -280,6 +280,30 @@ namespace ChimeraTK {
   : name(name_), nElements(nElements_), nChannels(nChannels_), is2DMultiplexed(is2DMultiplexed_), address(address_),
     nBytes(nBytes_), bar(bar_), width(width_), nFractionalBits(nFractionalBits_), signedFlag(signedFlag_),
     module(module_), registerAccess(dataAccess_), dataType(dataType_) {
+    if(nBytes_ > 0 && nElements_ > 0) {
+      if(nBytes_ % nElements_ != 0) {
+        // nBytes_ must be divisible by nElements_
+        throw logic_error(
+            "Number of bytes is not a multiple of number of elements for register " + name_ + ". Check your map file!");
+      }
+    }
+    DataType rawDataInfo;
+    if(is2DMultiplexed_) {
```

Comment:
Why do you tread multiplexed2D separately from else? They both set none.
(I don't disagree to do so. It's probably be a good idea to point this out explicitly.)

Reply:
It's not treated differently from the else, but from the 3 other branches. I could also write it like this:
```
f(nBytesPerElement() == 1 && !is2DMultiplexed_) {
      rawDataInfo = DataType::int8;
    }
    else if(nBytesPerElement() == 2 && !is2DMultiplexed_) {
      rawDataInfo = DataType::int16;
    }
    else if(nBytesPerElement() == 4) && !is2DMultiplexed_ {
      rawDataInfo = DataType::int32;
    }
    else {
      rawDataInfo = DataType::none;
    }
```

(Another question is why don't we support raw for multiplexed2D, but I did not want to touch this topic in this PR.)

Reply:
I have changed this in commit 81ecf4daf0b1733d643d4fcd8378db59c65aae6c

Reply:
Ah, now it is clear.

Reply:
I don't know if we need the lock:
Accessors are not thead-safe, and I think also TransferGrous and LowLevelTransferElements are not. So from that thread there is no concurrency.
From another thread with a different accessor and a different transfer element it not effective (never is, cannot be conceptually)

Reply:
I would have to read the spec again, but I think the low level transfer element is not allowed to read from the hardware in preRead(). But at the moment I see no other way.
Is it ensured that the low level pre-read is executed before the high-level pre-read?